src/draco/io/stl_decoder.cc

```diff
@@ -163,53 +174,85 @@ Status StlDecoder::ParseAsciiFace(Vector3f* v0, Vector3f* v1, Vector3f* v2, Vect
   return status;
 }
 
-Status StlDecoder::DecodeInternal() {
+Status StlDecoder::ParseAsAscii(bool* should_attempt_binary, std::vector<Vector3f>* norm_and_verts,
+                                uint32_t* num_faces) {
+  // For the ASCII formatted STL file we do not know how many triangles are specified in the file
+  // without reading the entire file.  We attempt to read as many ascii solids as possible.
+  // If we have an error while parsing the first ascii solid within the first 80 bytes, we
+  // assume the file is a binary stl and fall back to that.
   Vector3f tmp_norm;
   Vector3f tmp_v0;
   Vector3f tmp_v1;
   Vector3f tmp_v2;
+  bool is_first_solid = true;
+  bool is_valid_triangle;
+  Status status;
+  while (1) {
+    do {
+      status = ParseAsciiFace(&tmp_v0, &tmp_v1, &tmp_v2, &tmp_norm, &is_valid_triangle);
+      if (! status.ok()) {
+        *should_attempt_binary = is_first_solid;
```

Comment:
You never set this to false in this function. You do in the current calling function so it works for now but seems risky and might break easily after future code changes.